### PR TITLE
Use named casts instead of old-style casts

### PR DESCRIPTION
**Pull request by janisozaur**
Change prepared with clang-tidy and google-readability-casting check

### CODE REVIEW

**Comment by tupaschoal** [comment]
Should probably mention that it is part of #11158

**Comment by janisozaur** [comment]
@tupaschoal sure, forgot about that. But you can amend the commit message when you squash-merge it.

File: src/openrct2/actions/RideCreateAction.hpp [hunk 1]
```diff
@@ -178,7 +178,7 @@ DEFINE_GAME_ACTION(RideCreateAction, GAME_COMMAND_CREATE_RIDE, RideCreateGameAct
         ride->lift_hill_speed = RideTypeDescriptors[ride->type].LiftData.minimum_speed;
 
         ride->measurement = {};
-        ride->excitement = (ride_rating)-1;
+        ride->excitement = static_cast<ride_rating>(-1);
```
Comment (Gymnasiast) [inline src/openrct2/actions/RideCreateAction.hpp:181]:
```suggestion
        ride->excitement = RIDE_RATING_UNDEFINED;
```

File: src/openrct2/actions/RideEntranceExitPlaceAction.hpp [hunk 2]
```diff
@@ -200,12 +200,12 @@ DEFINE_GAME_ACTION(RideEntranceExitPlaceAction, GAME_COMMAND_PLACE_RIDE_ENTRANCE
         if (_isExit)
         {
             ride_set_exit_location(
-                ride, _stationNum, TileCoordsXYZD(CoordsXYZD{ _loc, z, (uint8_t)tileElement->GetDirection() }));
+                ride, _stationNum, TileCoordsXYZD(CoordsXYZD{ _loc, z, static_cast<uint8_t>(tileElement->GetDirection()) }));
```
Comment (Gymnasiast) [inline src/openrct2/actions/RideEntranceExitPlaceAction.hpp:203]: Cast should not be necessary.

```suggestion
                ride, _stationNum, TileCoordsXYZD(CoordsXYZD{ _loc, z, tileElement->GetDirection() }));
```

File: src/openrct2/actions/RideEntranceExitPlaceAction.hpp [hunk 3]
```diff
@@ -200,12 +200,12 @@ DEFINE_GAME_ACTION(RideEntranceExitPlaceAction, GAME_COMMAND_PLACE_RIDE_ENTRANCE
...
         }
         else
         {
             ride_set_entrance_location(
-                ride, _stationNum, TileCoordsXYZD(CoordsXYZD{ _loc, z, (uint8_t)tileElement->GetDirection() }));
+                ride, _stationNum, TileCoordsXYZD(CoordsXYZD{ _loc, z, static_cast<uint8_t>(tileElement->GetDirection()) }));
```
Comment (Gymnasiast) [inline src/openrct2/actions/RideEntranceExitPlaceAction.hpp:208]:
```suggestion
                ride, _stationNum, TileCoordsXYZD(CoordsXYZD{ _loc, z, tileElement->GetDirection() }));
```

**Comment by janisozaur** [comment]
I wish there was a away to select multiple suggestion to commit at once

**Comment by tupaschoal** [comment]
> I wish there was a away to select multiple suggestion to commit at once

There is
image

**Comment by janisozaur** [comment]
I… I saw this button and read the text it said, but my mind must've been turned off at the time.

File: src/openrct2/peep/Staff.cpp [hunk 4]
```diff
@@ -390,7 +390,7 @@ static bool staff_is_patrol_area_set(int32_t staffIndex, int32_t x, int32_t y)
     int32_t peepOffset = staffIndex * STAFF_PATROL_AREA_SIZE;
     int32_t offset = (x | y) >> 5;
     int32_t bitIndex = (x | y) & 0x1F;
-    return gStaffPatrolAreas[peepOffset + offset] & (((uint32_t)1) << bitIndex);
+    return gStaffPatrolAreas[peepOffset + offset] & ((static_cast<uint32_t>(1)) << bitIndex);
```
Comment (Gymnasiast) [inline src/openrct2/peep/Staff.cpp:393]:
```suggestion
    return gStaffPatrolAreas[peepOffset + offset] & (1UL << bitIndex);
```

File: src/openrct2/peep/Staff.cpp [hunk 5]
```diff
@@ -441,7 +441,7 @@ void staff_toggle_patrol_area(int32_t staffIndex, int32_t x, int32_t y)
  */
 static uint8_t staff_handyman_direction_to_nearest_litter(Peep* peep)
 {
-    uint16_t nearestLitterDist = (uint16_t)-1;
+    uint16_t nearestLitterDist = static_cast<uint16_t>(-1);
```
Comment (Gymnasiast) [inline src/openrct2/peep/Staff.cpp:444]:
```suggestion
    uint16_t nearestLitterDist = 0xFFFF;
```

File: src/openrct2/readme.md [hunk 6]
```diff
@@ -91,4 +91,3 @@
 - **world**
 
   World objects and mechanics such as the climate, landscape, sprites and park.
-
```
Comment (Gymnasiast) [inline src/openrct2/readme.md:94]: This doesn't really belong in this PR.
